arenadata/Dockerfile

```diff
@@ -1,45 +1,53 @@
-    yum clean all
+RUN dnf -y install 'dnf-command(config-manager)' && dnf config-manager --set-enabled devel && dnf makecache && dnf update -y ca-certificates && \
+    rpm --import https://mirror.yandex.ru/rockylinux/RPM-GPG-KEY-Rocky-8 && \
+    dnf -y install epel-release java-11-openjdk-devel && \
+    dnf -y install git iproute net-tools openssh-server rsync sudo time vim wget unzip \
+                   autoconf bison cmake3 flex gperftools indent jq libtool gcc-c++ \
+                   krb5-server krb5-workstation xerces-c-devel file netcat passwd lsof xz && \
+    dnf clean all
 
 # install all software we need
-RUN yum makecache && \
-    yum -y install python3-devel && \
-    yum -y install apr-devel bzip2-devel expat-devel libcurl-devel && \
-    yum -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
-    yum -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
-    yum -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
-    yum -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel && \
-    yum clean all
+RUN dnf makecache && \
+    dnf -y install python39-devel && \
+    dnf -y install apr-devel bzip2 bzip2-devel expat-devel libcurl-devel && \
+    dnf -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
+    dnf -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
+    dnf -y install libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
+    dnf -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel && \
+    dnf -y install python39-psycopg2 python39-psutil python39-pyyaml python39-lxml && \
+    dnf clean all && \
+    #we install pytest from pypi since the version from the repository does not contain an executable file
+    pip3 install pytest
+
+RUN dnf -y install \
+                --repofrompath=devel_external,https://dl.rockylinux.org/vault/rocky/8.8/devel/x86_64/os \
+                llvm-libs-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-test-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-static-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-devel-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                clang-resource-filesystem-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                clang-libs-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                clang-15.0.7-1.module+el8.8.0+1144+0a4e73bd&& \
+    dnf clean all
 
 # setup ssh configuration
 RUN ssh-keygen -t rsa -N "" -f /root/.ssh/id_rsa && \
     cat /root/.ssh/id_rsa.pub >> /root/.ssh/authorized_keys && \
     chmod 0600 /root/.ssh/authorized_keys && \
     echo -e "password\npassword" | passwd 2> /dev/null && \
-    { ssh-keyscan localhost; ssh-keyscan 0.0.0.0; } >> /root/.ssh/known_hosts && \
-    #
-    ssh-keygen -f /etc/ssh/ssh_host_key -N '' -t rsa1 && \
```

Comment:
Why was key generation removed?

Reply:
rsa1 is deprecated and an attempt to generate it fails.